File: src/test/java/tool/cmnclslib/cls/UnitTest_ClsJp1Job.java
package tool.cmnclslib.cls;

import org.junit.Before;
import org.junit.Test;
import static org.junit.Assert.*;

/**
 * ClsJp1Job の単体テストクラスです。
 */
public class UnitTest_ClsJp1Job {

    private ClsLogger logger = new ClsLogger();
    private ClsJp1Job jp1;

    @Before
    public void setUp() {
        jp1 = new ClsJp1Job(logger);
        jp1.setJobName("/ENV.PROD/JOB_SQLF.list_schema.sql__ID_A.10__ID_B.20.sq__ID_C.30.dq/I52.ADサーバ再起動/I22.再起動.RHOST.WEB-SV01..RTARGET.WEB-AP01/AAA/JOB_SQLF.replace.sql__BBB");
    }

    @Test
    public void 環境変数をセットできること() {
        assertTrue(jp1.setEnvironmentVariable());
    }

    @Test
    public void convertStringFromEnvironmentでENVを指定した場合値はPRODであること() {
        jp1.setEnvironmentVariable();
        assertEquals("PROD", jp1.convertStringFromEnvironment("AJSENV.ENV"));
    }

    @Test
    public void convertStringFromEnvironmentでID_Aを指定した場合値は10であること() {
        jp1.setEnvironmentVariable();
        assertEquals("10", jp1.convertStringFromEnvironment("AJSENV.ID_A"));
    }

    @Test
    public void convertStringFromEnvironmentでID_Bを指定した場合値は20であること() {
        jp1.setEnvironmentVariable();
        assertEquals("'20'", jp1.convertStringFromEnvironment("AJSENV.ID_B"));
    }

    @Test
    public void convertStringFromEnvironmentでID_Cを指定した場合値は30であること() {
        jp1.setEnvironmentVariable();
        assertEquals("\"30\"", jp1.convertStringFromEnvironment("AJSENV.ID_C"));
    }

    @Test
    public void convertStringFromEnvironmentでID_Bを指定した場合値はシングルクォーテーションなし20でないこと() {
        jp1.setEnvironmentVariable();
        assertNotEquals("20", jp1.convertStringFromEnvironment("AJSENV.ID_B"));
    }

    @Test
    public void convertStringFromEnvironmentでID_Cを指定した場合値はダブルクォーテーションなし30でないこと() {
        jp1.setEnvironmentVariable();
        assertNotEquals("30", jp1.convertStringFromEnvironment("AJSENV.ID_C"));
    }

    @Test
    public void convertStringFromEnvironmentでRHOSTを指定した場合値はWEBSV01であること() {
        jp1.setEnvironmentVariable();
        assertEquals("WEB-SV01", jp1.convertStringFromEnvironment("AJSENV.RHOST"));
    }

    @Test
    public void convertStringFromEnvironmentでRTARGETを指定した場合値はWEBAP01であること() {
        jp1.setEnvironmentVariable();
        assertEquals("WEB-AP01", jp1.convertStringFromEnvironment("AJSENV.RTARGET"));
    }

    @Test
    public void convertStringFromEnvironmentでJOB_SQLFを指定した場合値はreplacesqlであること() {
        jp1.setEnvironmentVariable();
        assertEquals("replace.sql", jp1.convertStringFromEnvironment("AJSENV.JOB_SQLF"));
    }

    @Test
    public void convertFromEnvで新設メソッドが正しく動作すること() {
        assertTrue(jp1.setEnvVariable());
        assertEquals("PROD", jp1.convertFromEnv("AJSENV.ENV"));
        assertEquals("10", jp1.convertFromEnv("AJSENV.ID_A"));
        assertEquals("'20'", jp1.convertFromEnv("AJSENV.ID_B"));
        assertEquals("\"30\"", jp1.convertFromEnv("AJSENV.ID_C"));
    }
}

File: TestProject1/Class/UnitTest_ClsJp1Job.cs
using CmnClsLib.Module;
using CmnClsLib.Class;
using CmnClsLib.Interface;
using Xunit;
using Assert = Xunit.Assert;

namespace TestProject1.Class
{
    public class UnitTest_ClsJp1Job
    {
        private static ClsLogger _logger = new();
        private ClsJp1Job _jp1 = new(_logger);

        public UnitTest_ClsJp1Job()
        {
            _jp1.JobName = "/ENV.PROD/JOB_SQLF.list_schema.sql__ID_A.10__ID_B.20.sq__ID_C.30.dq/I52.ADサーバ再起動/I22.再起動.RHOST.WEB-SV01..RTARGET.WEB-AP01/AAA/JOB_SQLF.replace.sql__BBB";
        }

        // --------------------------------------------------------------------
        // SetEnvironmentVariable()
        // --------------------------------------------------------------------
        [Fact]
        public void 環境変数をセットできること()
        {
            Assert.True(_jp1.SetEnvironmentVariable());
        }

        // --------------------------------------------------------------------
        // ConvertStringFromEnvironment()
        // --------------------------------------------------------------------
        [Fact]
        public void ConvertStringFromEnvironmentでENVを指定した場合値はPRODであること()
        {
            Assert.Equal("PROD", _jp1.ConvertStringFromEnvironment("AJSENV.ENV"));
        }
        [Fact]
        public void ConvertStringFromEnvironmentでID_Aを指定した場合値は10であること()
        {
            Assert.Equal("10", _jp1.ConvertStringFromEnvironment("AJSENV.ID_A"));
        }
        [Fact]
        public void ConvertStringFromEnvironmentでID_Bを指定した場合値は20であること()
        {
            Assert.Equal("'20'", _jp1.ConvertStringFromEnvironment("AJSENV.ID_B"));
        }
        [Fact]
        public void ConvertStringFromEnvironmentでID_Cを指定した場合値は30であること()
        {
            Assert.Equal("\"30\"", _jp1.ConvertStringFromEnvironment("AJSENV.ID_C"));
        }
        [Fact]
        public void ConvertStringFromEnvironmentでID_Bを指定した場合値はシングルクォーテーションなし20でないこと()
        {
            Assert.NotEqual("20", _jp1.ConvertStringFromEnvironment("AJSENV.ID_B"));
        }
        [Fact]
        public void ConvertStringFromEnvironmentでID_Cを指定した場合値はダブルクォーテーションなし30でないこと()
        {
            Assert.NotEqual("30", _jp1.ConvertStringFromEnvironment("AJSENV.ID_C"));
        }
        [Fact]
        public void ConvertStringFromEnvironmentでRHOSTを指定した場合値はWEBSV01であること()
        {
            Assert.Equal("WEB-SV01", _jp1.ConvertStringFromEnvironment("AJSENV.RHOST"));
        }
        [Fact]
        public void ConvertStringFromEnvironmentでRTARGETを指定した場合値はWEBAP01であること()
        {
            Assert.Equal("WEB-AP01", _jp1.ConvertStringFromEnvironment("AJSENV.RTARGET"));
        }
        [Fact]
        public void ConvertStringFromEnvironmentでJOB_SQLFを指定した場合値はreplacesqlであること()
        {
            Assert.Equal("replace.sql", _jp1.ConvertStringFromEnvironment("AJSENV.JOB_SQLF"));
        }
    }
}
